tests/test_fibers.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+// Based on test_coroutines.cpp
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <emscripten.h>
+
+struct Fiber {
+    emscripten_fiber context;
+    void *stack;
+    int result;
+
+    static void bad(void *arg) {
+        abort();
+    }
+
+    void init(em_arg_callback_func entry, void *arg) {
+        const int stack_size = 1 << 13;
+        stack = malloc(stack_size);
+        context = emscripten_fiber_create(bad, nullptr, stack, stack_size);
+        emscripten_fiber_recycle(context, entry, arg);
+        result = 0;
+    }
+
+    ~Fiber() {
+        free(stack);
+        emscripten_fiber_free(context);
+        printf("\ndestructor");
+    }
+};
+
+static struct Globals {
+    emscripten_fiber main;
+    Fiber fibers[2];
+
+    Globals() {
+        main = emscripten_fiber_create_from_current_context();
+    }
+
+    ~Globals() {
+        emscripten_fiber_free(main);
+        printf("\ndestructor");
+    }
```

Comment:
Can you just put these at the start and end of main() itself to keep the example simple?

Reply:
The test is also meant to make sure that the runtime shuts down correctly (with `EXIT_RUNTIME`). I can still put all the fiber initialization/destruction into main and keep a dummy destructor or `atexit` callback for that purpose, is that preferable?